zerver/worker/queue_processors.py

```diff
@@ -182,7 +181,6 @@ def wrapper(worker: ConcreteQueueWorker, data: Dict[str, Any]) -> None:
         try:
             func(worker, data)
         except (
-            smtplib.SMTPServerDisconnected,
             socket.gaierror,
             socket.timeout,
             EmailNotDeliveredException,
```

Comment:
I also think that it's a misleading for the commit message to say "In order to allow Zulip calling code to just catch a single exception" when we're still catching `socket.timeout` and `socket.gaierror`.  On the other hand, if we transform them into the plain `EmailNotDeliveredException`, we lose the additional detail of `error_class_name`, below.

So I think if we're going to do this change, we should back off the strength of the commit message to just be about simplifying the number of exceptions, not "just catch a single exception."

Reply:
I modified the commit message accordingly. Feel free to modify it further before merging.